google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -372,6 +375,16 @@ public StorageObject create(
     }
   }
 
+  private static String detectContentType(StorageObject object) {
```

Comment:
I was thinking about making this feature optional, but I didn't find an appropriate place to store this setting. It could be a property of a project/client/bucket/object.

I only found: https://cloud.google.com/storage/docs/metadata#content-type
which states:  `If the Content-Type is not specified by the uploader and cannot be determined, it is set to application/octet-stream or application/x-www-form-urlencoded, depending on how you uploaded the object.`

If I read this correctly, the Storage should attempt to determine the content type if not explicitly set, so this feature should be on by default.

BTW, right now the content type of blobs created with Storage.create() methods is `null`, but should be `application/octet-stream`.

Reply:
Storage doesn't provide contentType detection in the API.

If contentType is set at creation then `application/octet-stream` is used.

Please make this optional otherwise it's a behavior change not originally intended on uploads with the Java client.